src/backend/distributed/commands/foreign_constraint.c

```diff
@@ -561,106 +541,79 @@ get_relation_constraint_oid_compat(HeapTuple heapTuple)
 
 
 /*
- * HasForeignKeyToReferenceTable function scans the pgConstraint table to
- * fetch all of the constraints on the given relationId and see if at least one
- * of them is a foreign key referencing to a reference table.
+ * HasForeignKeyToReferenceTable function returns true if any of the foreign
+ * key constraints on the relation with relationOid references to a reference
+ * table.
  */
 bool
-HasForeignKeyToReferenceTable(Oid relationId)
+HasForeignKeyToReferenceTable(Oid relationOid)
 {
-	ScanKeyData scanKey[1];
-	int scanKeyCount = 1;
-	bool hasForeignKeyToReferenceTable = false;
+	bool findAll = false;
+	return ForeignKeysToReferenceTables(relationOid, findAll) != NIL;
```

Comment:
Maybe `return list_length(ForeignKeysToReferenceTables(relationOid, findAll)) > 0`

Reply:
isn't `!= NIL` much more readable and simple ?

Reply:
I think list_length > 0 is more robust because:
- it checks NIL
- it also checks if it has any element

Currently in terms of functionality it doesn't make difference.  So feel free to choose whichever you want to.

Reply:
now list_length makes more sense to me, I will use it instead, thanks!